lib/aws/xml.ex

```diff
@@ -0,0 +1,23 @@
+defmodule AWS.XML do
+  @moduledoc """
+  Specifies the behaviour of a XML parser.
+
+  You can switch the default XML parser which uses AWS.Util underneath
+  by defining a different implementation by setting the `:xml_module`
+  configuration in AWS.Client:
+
+     client = %AWS.Client{xml_module: MyCustomXMLParser}
+     Aws.SNS.publish(client, %{})
+
+  """
+
+  @doc """
+  Encodes a map into XML string. Raises in case of errors.
+  """
+  @callback encode!(input :: map(), options :: keyword()) :: String.t()
```

Comment:
Same thing, `encode_to_iodata!`. Strings are a subset of iodata, so we should be fine.

Reply:
Also took the opportunity to move the xml encoding and decoding from AWS.Util.